src/python/pants/java/nailgun_protocol.py

```diff
@@ -87,7 +87,10 @@ class TruncatedPayloadError(TruncatedRead):
   @classmethod
   def _decode_unicode_seq(cls, seq):
     for item in seq:
-      yield item.decode('utf-8')
+      if isinstance(item, bytes):
+        yield item.decode('utf-8')
+      else:
+        yield item
```

Comment:
This changes the semantics less than my earlier proposed change, and should still fix the `AttributeError` of calling `newstr.decode()`.

I'm still not sure the logic behind going from `List[bytes] or List[str] -> List[str] -> str -> bytes`, i.e. why we can't pass the bytes directly. But the test `test_pantsd_unicode_environment` from  `pants_test/pantsd/test_pantsd_integration.py:553` was failing when I tried to change this with e3877fa.